nmtwizard/preprocess/tu.py

```diff
@@ -93,6 +132,51 @@ def detok(self, detok):
         self.__detok = detok
         self.__tok = []
 
+
+    def replace_tokens(self, start_idx, tok_num, new_tokens=None, part=0):
+
+        # check/initialize tokenization if not done already
+        if self.tok.tokens:
+            if start_idx > len(self.__tok[part]):
+                raise RuntimeError('Start index is too big for replacement.')
```

Comment:
Maybe raise an `IndexError` here.

https://docs.python.org/3/library/exceptions.html#IndexError

Reply:
Done :)

Reply:
Could you also update the one just below?

Reply:
Done again :)